### PR TITLE
#317: Radio button component

### PR DESCRIPTION
**Pull request by BeeMargarida**
| - | - |
| --- | --- |
| Issue | https://github.com/ripe-tech/ripe-design/issues/317 |
| Dependencies | -- |
| Decisions | - Radio button <br> Uses the same label + value logic in components-vue, although I'm not sure if necessary. <br> - Ordered imports in `atoms/index.js`. <br> Same changes in `touchable` made in checkbox. |
| Animated GIF | radio|

### CODE REVIEW

File: react/components/atoms/index.js [hunk 1]
```diff
@@ -8,12 +8,12 @@ export * from "./button-tab";
 export * from "./button-tab-text";
 export * from "./container-draggable";
 export * from "./container-openable";
+export * from "./container-swipeable";
 export * from "./icon";
-export * from "./lightbox";
 export * from "./link";
+export * from "./lightbox";
```
Comment (beneditomauro) [inline react/components/atoms/index.js:14]: Alphabetically

File: react/components/atoms/radio/radio.js [hunk 2]
```diff
@@ -0,0 +1,182 @@
+import React, { PureComponent } from "react";
+import { Platform, StyleSheet, Text, View, ViewPropTypes } from "react-native";
+import PropTypes from "prop-types";
+import { mix } from "yonius";
+
+import { IdentifiableMixin, baseStyles, capitalize } from "../../../util";
+
+import { Touchable } from "../touchable";
+
+export class Radio extends mix(PureComponent).with(IdentifiableMixin) {
```
Comment (beneditomauro) [inline react/components/atoms/radio/radio.js:10]: Id the component

**Comment by beneditomauro** [comment]
Merge with master

**Review by beneditomauro — APPROVED** [review]

File: react/components/atoms/radio/radio.stories.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import { storiesOf } from "@storybook/react-native";
+import { withKnobs, boolean, number, select, text } from "@storybook/addon-knobs";
+
+import { Radio } from "./radio";
+
+storiesOf("Components/Atoms/Radio", module)
+    .addDecorator(withKnobs)
+    .add("Radio", () => {
+        const label = text("Label", "Label");
+        const value = text("Value", undefined);
```
Comment (gcandal) [inline react/components/atoms/radio/radio.stories.js:11]: Why the different approach to defaulting?

File: react/components/atoms/radio/radio.js [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+    static get propTypes() {
+        return {
+            label: PropTypes.string,
+            value: PropTypes.any,
+            checked: PropTypes.bool,
+            disabled: PropTypes.bool,
+            variant: PropTypes.string,
+            size: PropTypes.number,
+            onUpdateChecked: PropTypes.func,
+            style: ViewPropTypes.style
```
Comment (gcandal) [inline react/components/atoms/radio/radio.js:20]: Does this respect the usual ordering between event handling + style props?

Comment (BeeMargarida) [reply react/components/atoms/radio/radio.js:20]: Yap, in the majority of the components style is always at the bottom (in a small number of components its not, but I'll make a PR fixing it).
In `sdk-components-react` there is only one component, and its different: https://github.com/ripe-tech/ripe-sdk-components-react/blob/47bdc3fd91af19ff74ee69783d183a75495238a9/react/components/organisms/ripe-image/ripe-image.js#L253-L257. Do I change it?

Comment (gcandal) [reply react/components/atoms/radio/radio.js:20]: Yep, go ahead. Thanks for keeping things tidy 🙏

**Review by gcandal — APPROVED** [review]

**Review by joamag — APPROVED** [review]

**Pull request merged**